src/core/constructs/workspace_watcher.py

```diff
@@ -0,0 +1,84 @@
+import time
+from typing import Optional
+
+from watchdog.tricks import Trick
+
+from core.constructs.workspace import Workspace
+from core.constructs.output_manager import OutputManager
+from core.commands.deploy_differences import execute_deployment
+from watchdog.observers import Observer
+
+
+class WorkspaceWatcher(Trick):
+    """Executes some commands in response to modified files."""
+
+    _default_patterns_to_watch = ["src/**/*.py", "settings/*"]
+    _default_patterns_to_ignore = [".cdev/**", "__pycache__/*"]
+
+    def __init__(
+        self,
+        workspace: Workspace,
+        output: OutputManager,
+        no_default: Optional[bool] = False,
+        patterns_to_watch: Optional[str] = None,
+        patterns_to_ignore: Optional[str] = None
+    ) -> None:
+
+        self._observer = None
+        self._workspace = workspace
+        self._output = output
+        self._base_folder = self._workspace.settings.BASE_PATH
+        self._perform_deployment = False
+
+        all_patterns_to_watch = None
```

Comment:
Good catch!
I've added the no-default tested it but then I cleared the code and didn't see this path.
I'm not initializing it to empty because that means that the pattern will not match anything. It has to be None